dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -0,0 +1,353 @@
+    val didObj = DID.fromString(did)
+    require(didObj.methodName == methodName)
+
+    val resp = runBlocking { client.get("$identifiersEndpoint/$didObj") }
+    val body = runBlocking { resp.bodyAsText() }
+    if (!resp.status.isSuccess()) {
+      throw InvalidStatusException("resolution error response '$body'")
+    }
+    return mapper.readValue(body, DidResolutionResult::class.java)
+  }
+
+  private fun createOperation(keyManager: KeyManager, options: CreateDidIonOptions?)
+    : Pair<SidetreeCreateOperation, KeyAliases> {
+    val updatePublicJWK: JWK = if (options?.updatePublicJWK == null) {
+      val alias = keyManager.generatePrivateKey(JWSAlgorithm.ES256K, Curve.SECP256K1)
+      keyManager.getPublicKey(alias)
+    } else {
+      options.updatePublicJWK
+    }
+    val publicKeyCommitment: String = publicKeyCommitment(updatePublicJWK)
+
+    val verificationMethodId = options?.verificationMethodId ?: UUID.randomUUID().toString()
+    validateVerificationMethodId(verificationMethodId)
+    val verificationPublicKey = if (options?.verificationPublicKey == null) {
+      val alias = keyManager.generatePrivateKey(JWSAlgorithm.ES256K, Curve.SECP256K1)
+      val verificationJWK = keyManager.getPublicKey(alias)
+      PublicKey(
+        id = verificationMethodId,
+        type = "JsonWebKey2020",
+        publicKeyJwk = verificationJWK,
+        purposes = listOf(PublicKeyPurpose.AUTHENTICATION),
+      )
+    } else {
+      options.verificationPublicKey
+    }
+    val patches = listOf(ReplaceAction(Document(listOf(verificationPublicKey))))
+    val createOperationDelta = Delta(
+      patches = patches,
+      updateCommitment = publicKeyCommitment
+    )
+
+    val recoveryPublicJWK = if (options?.recoveryPublicJWK == null) {
+      val alias = keyManager.generatePrivateKey(JWSAlgorithm.ES256K, Curve.SECP256K1)
+      keyManager.getPublicKey(alias)
+    } else {
+      options.recoveryPublicJWK
+    }
+    val recoveryCommitment = publicKeyCommitment(recoveryPublicJWK)
+
+    val operation: OperationSuffixDataObject =
+      createOperationSuffixDataObject(createOperationDelta, recoveryCommitment)
+
+    return Pair(
+      SidetreeCreateOperation(
+        type = "create",
+        suffixData = operation,
+        delta = createOperationDelta,
+      ),
+      KeyAliases(
+        updateKeyAlias = updatePublicJWK.keyID,
+        verificationKeyAlias = verificationPublicKey.publicKeyJwk!!.keyID,
+        recoveryKeyAlias = recoveryPublicJWK.keyID
+      )
+    )
+  }
+
+  private fun validateVerificationMethodId(id: String) {
+    if (!isBase64UrlString(id)) {
+      throw IllegalArgumentException("verification method id \"$id\" is not base 64 url charset")
+    }
+
+    if (id.length > maxVerificationMethodIdLength) {
+      throw IllegalArgumentException(
+        "verification method id \"$id\" exceeds max allowed length of $maxVerificationMethodIdLength"
+      )
+    }
+  }
```

Comment:
is this a sidetree requirement? the did core spec states otherwise

![image](https://github.com/TBD54566975/web5-kt/assets/4887440/cfa001ee-a0f3-4455-919f-c556b152ab1c)


with the option of an id simply being `#fragment`.